src/plugins/auto/executable_network.cpp

```diff
@@ -608,7 +646,19 @@ InferenceEngine::Parameter MultiDeviceExecutableNetwork::GetConfig(const std::st
 
 InferenceEngine::Parameter MultiDeviceExecutableNetwork::GetMetric(const std::string &name) const {
     if (_workModeIsAUTO) {
-        // fixme: should we wait actual device? meanwhile it will block inference, how to fix?
+        if (name == METRIC_KEY(OPTIMAL_NUMBER_OF_INFER_REQUESTS)) {
+            unsigned int real = 0;
+            if (_loadContext[ACTUALDEVICE].isAlready) {
+                real = _loadContext[ACTUALDEVICE].
+                    executableNetwork->GetMetric(name).as<unsigned int>();
+            } else {
```

Comment:
I would add a comment (or maybe also an assert) that the CPU's isAlready is true.
(it should, otherwise the Auto::LoatNetwork hasn't returned and no GetMetric call for the resulted network would be valid).

Reply:
yes, it's good for all APIs to add this assert if need